contracts/orbital-auction/src/state.rs

```diff
@@ -0,0 +1,266 @@
+}
+
+impl UserIntent {
+    /// splits the order into two orders, one with the given amount and the remainder.
+    /// returns an error if the amount exceeds the order amount. if it doesn't, returns
+    /// a tuple in the form of (new_order, remainder).
+    pub fn split_order(&self, amount: Uint128) -> StdResult<(UserIntent, UserIntent)> {
+        let new_order = UserIntent {
+            user: self.user.clone(),
+            amount,
+            offer_domain: self.offer_domain.clone(),
+            ask_domain: self.ask_domain.clone(),
+        };
+
+        let remainder = UserIntent {
+            user: self.user.clone(),
+            amount: self.amount.checked_sub(amount)?,
+            offer_domain: self.offer_domain.clone(),
+            ask_domain: self.ask_domain.clone(),
+        };
+
+        Ok((new_order, remainder))
+    }
+}
+
+#[cw_serde]
+pub struct AuctionConfig {
+    // how many of the offer denom we can fit in a batch
+    pub batch_size: Uint128,
+    // config that describes the time durations for each phase of the auction
+    pub auction_phases: AuctionPhaseConfig,
+    // config that describes the route for the auction
+    // (src & dest domains, offer & ask denoms)
+    // TODO: is this relevant?
+    pub route_config: RouteConfig,
+    // configured bond amount required to participate in the auction
+    pub solver_bond: Coin,
+}
+
+/// orbital auction operates in discrete-time based rounds.
+/// each round consists of the following phases, in order:
+/// - auction: bidding window where solvers can submit their bids
+/// - filling: window where the auction is finalized and orders are matched
+/// - cleanup: window where the auction is reset and the next round is prepared
+// TODO: validate that all durations are passed in seconds.
+// or just drop Duration altogether and deal in seconds?
+#[cw_serde]
+pub struct AuctionPhaseConfig {
+    // duration of the bidding window in seconds
+    pub auction_duration: Duration,
+    // duration of the filling window in seconds
+    pub filling_window_duration: Duration,
+    // duration of the cleanup window in seconds
+    pub cleanup_window_duration: Duration,
+}
+
+impl AuctionPhaseConfig {
+    /// validates that the auction phase configuration is all based on time (seconds).
+    pub fn validate(&self) -> StdResult<()> {
+        if let Duration::Height(_) = self.auction_duration {
+            return Err(StdError::generic_err(
+                "auction duration must be a time duration",
+            ));
+        }
+        if let Duration::Height(_) = self.filling_window_duration {
+            return Err(StdError::generic_err(
+                "filling duration must be a time duration",
+            ));
+        }
+        if let Duration::Height(_) = self.cleanup_window_duration {
+            return Err(StdError::generic_err(
+                "cleanup duration must be a time duration",
+            ));
+        }
+
+        Ok(())
+    }
+}
```

Comment:
Since we are using Timestamps only for everything, why not use u64 directly instead of using Duration?